### PR TITLE
new: retry on common errors

### PR DESCRIPTION
**Pull request by jriddle-linode**
## 📝 Description

**What does this PR do and why is this change necessary?**

Adding retry functionality for common random errors sent by the API

- 400: Bad Request with nginx headers
- 408: Request Timeout
- 429: Rate limit exceeded

## ✔️ How to Test

**What are the steps to reproduce the issue or verify the changes?**

**How do I run the relevant unit/integration tests?**

`make test`

### CODE REVIEW

**Review by ykim-1 — APPROVED** [review]
Looks good!

**Comment by zliang-akamai** [comment]
In Python SDK we used [the retry class from urllib3](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry). Should we consider that for linode-cli as well or?

File: linodecli/api_request.py [hunk 1]
```diff
@@ -339,3 +345,33 @@ def _handle_error(ctx, response):
...
+    return False
+
+
+def _get_retry_after(headers):
+    if headers.get("Retry-After"):
+        retry_str = headers.get("Retry-After")
+        if retry_str == "":
+            return 0
+        return int(retry_str)
+    return 0
```
Comment (amisiorek-akamai) [inline linodecli/api_request.py:377]: Looks good! Just a small suggestion.

```suggestion
    retry_str = headers.get("Retry-After", "")
    return int(retry_str) if retry_str else 0
```

**Comment by lgarber-akamai** [comment]
> In Python SDK we used [the retry class from urllib3](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry). Should we consider that for linode-cli as well or?
> 
> https://github.com/linode/linode_api4-python/blob/dev/linode_api4/linode_client.py#L26

That's a great question! The biggest limitation I can think of with that approach is that it might be a good bit more complicated to implement complex retry logic (e.g. 400 NGINX errors)

File: linodecli/api_request.py [hunk 2]
```diff
@@ -339,3 +345,33 @@ def _handle_error(ctx, response):
...
+        return True
+    if response.headers:
+        if (
+            response.status_code == 400
+            and response.headers.get("Server") == "nginx"
+            and response.headers.get("Content-Type") == "text/html"
+        ):
+            # nginx html response
+            return True
+    return False
```
Comment (lgarber-akamai) [inline linodecli/api_request.py:368]: Just a tiny optional code suggestion

```suggestion
    if response.status_code in (408, 429):
        # request timed out or rate limit exceeded
        return True

    if (
        response.headers 
        and response.status_code == 400
        and response.headers.get("Server") == "nginx"
        and response.headers.get("Content-Type") == "text/html"
    ):
        # nginx html response
        return True

    return False
```

**Review by lgarber-akamai — APPROVED** [review]
Looks good to me! Just needs a `make format` run .

**Pull request merged**